subprojects/core/src/main/java/org/gradle/initialization/DefaultSettingsLoader.java

```diff
@@ -173,17 +181,19 @@ private void validate(SettingsInternal settings) {
                 throw new GradleException("'" + SettingsInternal.BUILD_SRC + "' cannot be used as a project name as it is a reserved name" + suffix);
             }
             if (!project.getProjectDir().exists() || !project.getProjectDir().isDirectory() || !project.getProjectDir().canWrite()) {
-                emitProjectDirectoryMissingWarning(project.getPath(), project.getProjectDir().toString());
+                failOnMissingProjectDirectory(project.getPath(), project.getProjectDir().toString());
             }
         });
     }
 
-    private static void emitProjectDirectoryMissingWarning(String projectPath, String projectDir) {
-        String template = "Configuring project '%s' without an existing directory is deprecated. The configured projectDirectory '%s' does not exist, can't be written to or is not a directory.";
-        DeprecationLogger.deprecateBehaviour(String.format(template, projectPath, projectDir))
-            .withAdvice("Make sure the project directory exists and can be written.")
-            .willBecomeAnErrorInGradle9()
-            .withUpgradeGuideSection(8, "deprecated_missing_project_directory")
-            .nagUser();
+    private void failOnMissingProjectDirectory(String projectPath, String projectDir) {
+        String template = "Configuring project '%s' without an existing directory is not allowed. The configured projectDirectory '%s' does not exist, can't be written to or is not a directory.";
+        throw problems.getInternalReporter().throwing(
+            new GradleException(String.format(template, projectPath, projectDir)),
+            ProblemId.create("confituring-project-with-invalid-directory", "Configuring project with invalid directory", GradleCoreProblemGroup.configurationUsage()),
+            spec ->
+                spec.solution("Make sure the project directory exists and is writable.")
+                    .documentedAt(upgradeGuide(8, "deprecated_missing_project_directory").getUrl())
```

Comment:
❓ Shouldn't this be documented in the actual user guide and this reported problem link to that instead of linking to the upgrading guide?

Reply:
Maybe :).

I'm not sure where would be a good place for this.

@lkasso where would you put the doc for this?

Reply:
I'm going to write a separate PR to add a section in: https://docs.gradle.org/current/userguide/directory_layout.html. No need to change/add anything in this PR. @reinsch82, I will add you as a reviewer.